File: legacy-python/data_structures/dispute.py
class Dispute:
    def __init__(
        self,
        dao,
        parties_involved,
        description,
        importance=1,
        project=None,
        member=None,
    ):
        self.dao = dao
        self.parties_involved = parties_involved
        self.description = description
        self.importance = importance
        self.project = project
        self.member = member
        self.resolution = None
        self.resolved = False

    def resolve(self, resolution):
        self.resolution = resolution
        self.resolved = True

File: lib/data-structures/dispute.ts
// Dispute data structure

import type { DAO } from './dao';
import type { Project } from './project';

export class Dispute {
  dao: DAO;
  partiesInvolved: string[]; // Array of member unique IDs
  description: string;
  importance: number;
  project: Project | null;
  member: string | null; // Member unique ID
  resolution: string | null = null;
  resolved: boolean = false;
  uniqueId: string = '';

  constructor(
    dao: DAO,
    partiesInvolved: string[],
    description: string,
    importance: number = 1,
    project: Project | null = null,
    member: string | null = null
  ) {
    this.dao = dao;
    this.partiesInvolved = partiesInvolved;
    this.description = description;
    this.importance = importance;
    this.project = project;
    this.member = member;
  }

  resolve(resolution: string): void {
    this.resolution = resolution;
    this.resolved = true;
  }
}
